src/components/mfa/mfa-enroll.js

```diff
@@ -52,23 +54,43 @@ class MfaEnrollComponent {
 
   showComplete = () => {
     this.section = 'complete';
+    this.selectedFactor.isSubmitting = false;
     if (this.onComplete) {
       this.onComplete(this);
     }
   }
 
+
   showError = (err) => {
     if (err.error === 'invalid_request') {
       err.message = 'The code you entered was not valid.';
     }
 
+    if (err.message === 'An existing phone with that number is already associated with a factor for that Account.') {
```

Comment:
I agree on that it did feel dirty. But unfortunately all we get back is a 400 Bad Request and no error type. And I feel better about ug string matching than treating all as 400s as the same error...

Maybe we should we suggest this error message to be changed in the API, or that we add a error type to the request?

```json
{"status":400,"message":"An existing phone with that number is already associated with a factor for that Account."}
```